### PR TITLE
HV-1164 Only cast dynamicPayload if it is != null and add a test case

### PR DESCRIPTION
**Pull request by bayerls**

### CODE REVIEW

**Comment by gsmet** [comment]
Jenkins, add to whitelist

File: engine/src/main/java/org/hibernate/validator/internal/engine/ConstraintViolationImpl.java [hunk 1]
```diff
@@ -228,7 +228,7 @@ public Object getExecutableReturnValue() {
 
 	@Override
 	public <C> C getDynamicPayload(Class<C> type) {
-		if ( type.isAssignableFrom( this.dynamicPayload.getClass() ) ) {
+		if ( dynamicPayload != null && type.isAssignableFrom( this.dynamicPayload.getClass() ) ) {
```
Comment (gsmet) [inline engine/src/main/java/org/hibernate/validator/internal/engine/ConstraintViolationImpl.java]: Looks very good, thanks!

Minor comment: could you remove the `this.` occurrences in front of `dynamicPayload` in the existing code of this method and amend your commit? We should be consistent and I see no reason to keep `this.`.

Additional bonus point for being in line with our checkstyle rules!

**Comment by gsmet** [comment]
OK, perfect! Will squash and merge. Thanks!

**Comment by gsmet** [comment]
@bayerls one last thing: could you sign the CLA for the Hibernate Validator project at https://cla.jboss.org/index.seam? It's a one time thing for all your contributions to the project. 

Thanks!

**Comment by gsmet** [comment]
@bayerls can you ping me when it's done and give me your username? Thanks!

**Comment by bayerls** [comment]
@gsmet I signed and verified with the username bayerls some hours ago. Currently, the status is still "Awaiting approval by project lead". Glad I could help!

**Comment by gunnarmorling** [comment]
Thanks, @bayerls! The CLA is approved an I've merged your change into master (6.0) and 5.4.

**Comment by gunnarmorling** [comment]
Also added one more commit for some related clean-up.

**Pull request closed without merging**